compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/AggregatedOpInterfaceImpl.cpp

```diff
@@ -246,18 +326,69 @@ OnlineAttentionOp::decomposeOperation(OpBuilder &b) {
   AffineMap sumMap = getSumMap();
   Value normSum = scaleValueInPlace(b, loc, sumMap, normMap, oldSum, norm);
 
-  // newSum = normSum + rowMax(P)
+  // newSum = normSum + rowSum(P)
   Value newSum = reduce<arith::AddFOp>(b, loc, pMap, sumMap, p, normSum);
 
   // newAcc = norm * oldAcc
   AffineMap accMap = getOutputMap();
+
+  // ---- Scale and truncate LHS to match RHS ----
+  Value pScale;
+  auto pETy = getElementTypeOrSelf(p.getType());
+  if (pETy != vETy && isa<FloatType>(vETy)) {
+    if (vETy.getIntOrFloatBitWidth() <= 8) {
+      SmallVector<OpFoldResult> mSizes;
+      for (AffineExpr dimExpr : maxMap.getResults()) {
+        int dim = cast<AffineDimExpr>(dimExpr).getPosition();
```

Comment:
NIT: Any thoughts on using llvm::map_range?

Reply:
I changed to using `llvm::map_range`.